tornjak-frontend/src/components/types.ts

```diff
@@ -0,0 +1,99 @@
+  spiffeid: string; // SPIFFEE ID of agent
+  plugin: string; // Workload attestor plugin selected for agent
+  cluster: string; // Cluster agent is associated with
+}
+
+
+// clusters
+export interface ClustersListType {
+  name: string; // Name of Cluster
+  editedName: string; // Edited Name if Cluster Name is edited from original
+  creationTime: string; // Time cluster is created
+  domainName: string; // Domain Name of cluster if any
+  managedBy: string; // Person/ entity managing the cluster
+  platformType: string; // Platform type of the cluster
+  agentsList: Array<string>; // List of agents associated with the cluster
+}
+
+// entries
+export interface EntriesListType {
+  // From https://github.com/spiffe/spire-api-sdk/blob/main/proto/spire/api/types/entry.pb.go
+  id: string; // Globally unique ID for the entry.
+  spiffe_id: SPIFFEIDType; // The SPIFFE ID of the identity described by this entry.
+  // Who the entry is delegated to. If the entry describes a node, this is
+	// set to the SPIFFE ID of the SPIRE server of the trust domain (e.g.
+	// spiffe://example.org/spire/server). Otherwise, it will be set to a node
+	// SPIFFE ID.
+  parent_id: SPIFFEIDType;
+  // The selectors which identify which entities match this entry. If this is
+	// an entry for a node, these selectors represent selectors produced by
+	// node attestation. Otherwise, these selectors represent those produced by
+	// workload attestation.
+  selectors: Array<SelectorType>;
+  ttl: number; // The time to live for identities issued for this entry (in seconds).
+  federates_with: string[]; // The names of trust domains the identity described by this entry federates with
+  // Whether or not the identity described by this entry is an administrative
+	// workload. Administrative workloads are granted additional access to
+	// various managerial server APIs, such as entry registration.
+  admin: boolean;
+  // Whether or not the identity described by this entry represents a
+	// downstream SPIRE server. Downstream SPIRE servers have additional access
+	// to various signing APIs, such as those used to sign X.509 CA
+	// certificates and publish JWT signing keys.
+  downstream: boolean;
+  expires_at: number; // When the entry expires (seconds since Unix epoch).
+  dns_names: string[]; // A list of DNS names associated with the identity described by this entry.
+  revision_number: number; // Revision number is bumped every time the entry is updated
+  store_svid:  boolean; // Determines if the issued identity is exportable to a store
+}
+
+// servers
+export interface TornjakServerInfoType {
+  // Plugins for tornjak server info
+  plugins: {
+      DataStore: string[];
+      KeyManager: string[];
+      NodeAttestor: string[];
+      NodeResolver: string[];
+      Notifier: string[];
+  };
+  trustDomain: string; // Trust domain of server
+  verboseConfig: string; // More detailed metadata on server
+}
+
+export interface ServerInfoType {
+  data: {
```

Comment:
looks like its worthwhile to change this to just be flat

Reply:
what do you mean by flat?

Reply:
Resolved